src/main/java/de/flapdoodle/embed/process/store/LocalArtifactStore.java

```diff
@@ -38,16 +40,20 @@ public static boolean checkArtifact(DownloadConfig runtime, Distribution distrib
 		return getArtifact(runtime, distribution) != null;
 	}
 
-
 	public static boolean store(DownloadConfig runtime, Distribution distribution, File download) {
 		File dir = createOrGetBaseDir(runtime);
 		String artifactFileName = runtime.getPackageResolver().packageFor(distribution).archivePath();
 		File artifactFile = new File(dir, artifactFileName);
 		createOrCheckDir(artifactFile.getParentFile());
-		if (!Files.moveFile(download, artifactFile))
-			throw new IllegalArgumentException("Could not move " + download + " to " + artifactFile);
+		try {
+			Files.moveFile(download, artifactFile);
+		} catch (FileAlreadyExistsException faex) {
```

Comment:
We could also write over the existing file with CopyOption REPLACE_EXISTING. It would require an extra parameter on wrapper method Files.moveFile but it's another alternative.

I went with this solution since I supposed we do not want to replace this file if it exists.
Let me know if you disagree.